app/Exceptions/Handler.php

```diff
@@ -40,10 +44,31 @@ public function report(Exception $exception)
      *
      * @param  \Illuminate\Http\Request  $request
      * @param  \Exception  $exception
-     * @return \Illuminate\Http\Response
+     * @return \Illuminate\Http\Response|Response
      */
     public function render($request, Exception $exception)
     {
+        if ($exception instanceof TokenMismatchException) {
+            // Explicitly redirect to login form if token mismatches.
+            // Note: this will also draw attention to the login form timing out.
+            Auth::logout();
+            return redirect()
+                ->guest(route('service.login'))
+                // We anticipate expiry to be the most common reason.
+                ->withErrors(['error_message' => trans('auth.expired')])
+                ;
+        }
+
+        if ($exception instanceof MethodNotAllowedHttpException) {
+            // User somehow tried an HTTP verb a resource doesn't support
+            // Send them somewhere safe.
+            $route = (Auth::check()) ? 'service.dashboard' : 'service.login';
+            return redirect()
+                // For some reason ->withErrors() won't "take";
+                // The session gets flashed, but it doesn't survive.
```

Comment:
I guess that's not the end of the world. Worth keeping note here in case we figure out why.